### DISCUSSION TITLE
Linux ARM64 V6.0 test build

### DISCUSSION THREAD

**Discussion by gusmanb** [gusmanb/logicanalyzer#147] [General]
For ARM64 Linux users that would like to test the new V6.0 and be part of the debugging, [here is the current build](https://mega.nz/file/KXZg3QCb#SNABe2jyNOS7StIgitMd0Zw4R9pIgcRpInu0kjKWBE4).

Remember to use one of the [new firmwares](https://mega.nz/file/3SZCDQzD#krl3m4G7HKwNuCi6rsmw_yeyi9CK_usz6gGNyXah-uY).

Python 3.x must be installed on the system, minimum version 3.7. If the analyzer is not able to locate the version you can add a "python.cfg" file in the app folder containing the path to the "libpython3.x.so".

If you find bugs or errors please notify them here.

## Keep in mind!

This is a test build and to ease things I will upload multiple versions with updates answering to the posts with bugs, please read the thread to find which version is the latest one.

Cheers.

**Comment by DeeJay** [comment]
Briefly checked the arm64 version timestamped 'Sep 19 07:11' 

Host HW: RPi[5]-8G, OS: 64-bit RasPiOS Bookworm.
Client FW: standard .uf2 on a basic Pico

Data capture works. Protocol decoder for a 9600 baud serial UART works.

Suggestion: [Help]/[About] on the LA Desktop GUI still reports v5.1, so I can't accurately report which LA build I am running.

**Comment by DeeJay** [comment]
I have a 'Sticky Selection' issue similar to the one reported for the Colour Picker.

This one can be triggered by the 'Device Information'  **( i )** Dialogue.

To reproduce -
Start LA 6.0
In the top toolbar, confirm that [File] and [Help/About] perform as expected. (That's the 'Control' case.)

Select a Pico client device and Open the connection to it.
A summary description the device/firmware appears, followed by ( i ) . Clicking that icon gives a useful Device Information summary. 
Close that Device Information box.

Observe that clicking on [File] or [Help] or anywhere on the screen now results in the Device Information screen re-appearing.

Only way to exit the App is the Top-Right **'X'**

**Reply by gusmanb** [reply]
Yeah, this is a general problem in Avalonia with the dialogs and events in Linux... I will try to implement a generalized workaround but is going to take some time.

**Comment by gusmanb** [comment]
Latest link to folder with all the latest versions: https://mega.nz/folder/SGxDHAZL#afLGgQbJAaqOYXjhJBwokQ

**Reply by DeeJay** [reply]
Thank you. This build seems to have resolved the 'persistent menus' issue in respect of the Color Picker and the Board Information.

Since it is a development release it would be unrealistic to expect everything to work perfectly, so as requested this is feedback.

Access to a wirelessly-configured PicoW is not working with this build.

I have used the PicoW_WIFI variant of the firmware; configured network details; confirmed that my PicoW associates with my network and that port 4045 is Open. When trying to Open the Network device from LA6.0 I get the error -
Cannot connect to device: (Invalid device buffer size response.).

No urgency for a solution, just posting for information.

**Reply by gusmanb** [reply]
Have you updated the W firmware with the ones on the firmware folder on that link?

The protocol has changed a lot in the latest version, so if you have a previous one it is not compatible.

**Reply by DeeJay** [reply]
A reasonable question to check.

The answer is YES.  Audit trail follows..

```
david@rp58book101064fnvme:~/Projects/gusman/LA6b2fw $ 
david@rp58book101064fnvme:~/Projects/gusman/LA6b2fw $ 
david@rp58book101064fnvme:~/Projects/gusman/LA6b2fw $ ls -al logic-analyzer-firmware_v6.0.0_beta2_BOARD_PICO_W_WIFI.uf2 
-rw-r--r-- 1 david david 702464 Oct 12 11:58 logic-analyzer-firmware_v6.0.0_beta2_BOARD_PICO_W_WIFI.uf2
david@rp58book101064fnvme:~/Projects/gusman/LA6b2fw $ 
david@rp58book101064fnvme:~/Projects/gusman/LA6b2fw $ 
david@rp58book101064fnvme:~/Projects/gusman/LA6b2fw $ picotool info logic-analyzer-firmware_v6.0.0_beta2_BOARD_PICO_W_WIFI.uf2 
File logic-analyzer-firmware_v6.0.0_beta2_BOARD_PICO_W_WIFI.uf2:

Program Information
 name:          LogicAnalyzer
 version:       6.0
 features:      USB stdin / stdout
 binary start:  0x10000000
 binary end:    0x10055b64
david@rp58book101064fnvme:~/Projects/gusman/LA6b2fw $ 
david@rp58book101064fnvme:~/Projects/gusman/LA6b2fw $ 
david@rp58book101064fnvme:~/Projects/gusman/LA6b2fw $ 
david@rp58book101064fnvme:~/Projects/gusman/LA6b2fw $ picotool info -f
Tracking device serial number E6616407E346BC28 for reboot
The device was asked to reboot into BOOTSEL mode so the command can be executed.

Program Information
 name:          LogicAnalyzer
 version:       6.0
 features:      USB stdin / stdout
 binary start:  0x10000000
 binary end:    0x10055b64

                The device was asked to reboot back into application mode.
david@rp58book101064fnvme:~/Projects/gusman/LA6b2fw $ 
david@rp58book101064fnvme:~/Projects/gusman/LA6b2fw $ 

```


I also made sure of a clean installation by using 'flash_nuke.uf2' prior to installing the LA6b2 firmware.

**Reply by gusmanb** [reply]
Ok, there was a bug in the driver, it was missing the blast mode identification and it was treating the device's response as if it was the buffer size.

I have updated the folder with beta 3, the same link as before.

Cheers.

**Reply by DeeJay** [reply]
I can confirm that with LAv6 beta3 (App version string ending ..53270) the LA App for linux arm64 can connect wirelessly to a PicoW running the LAv6b2 Pico_W_WIFI firmware having previously configured the association with my home AP.

Live data acquisition and decoding untested at this time.
